### PR TITLE
Add install code to enable testing of OpenStack migration

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

File: cookbooks/calico/recipes/control.rb [hunk 1]
```diff
@@ -372,12 +383,29 @@
     notifies :create, "template[/etc/nova/nova.conf]", :immediately
     notifies :run, "execute[remove-old-nova-db]", :immediately
 end
+template "/etc/nova/nova.conf" do
+    mode "0640"
+    source "control/nova.conf.erb"
+    variables({
+        admin_password: node[:calico][:admin_password],
+        live_migreate: node[:calico][:live_migrate]
```
Comment (alexaltair) [inline cookbooks/calico/recipes/control.rb]: s/live_migreate:/live_migrate:

File: cookbooks/calico/recipes/control.rb [hunk 2]
```diff
@@ -460,6 +481,19 @@
...
 
+# Output and store the UID and GID for nova - this may be required for live migration
+execute "get-nova-info" do
+    command "id nova >> /tmp/nova.user"
+end
+ruby_block "store-nova-user-info" do
+    block do
+        output = ::File.read("/tmp/nova.user")	
+        match = /uid=(?<uid>\d+).*gid=(?<gid>\d+).*/.match(output)
+	node.set["nova_uid"] = match[:uid]
```
Comment (alexaltair) [inline cookbooks/calico/recipes/control.rb]: These two lines start with hard tabs.

Comment (alexaltair) [inline cookbooks/calico/recipes/control.rb]: Ack, actually lots of lines do; maybe your editor doesn't make them into spaces for Ruby?

File: cookbooks/calico/recipes/control.rb [hunk 3]
```diff
@@ -372,12 +383,29 @@
...
+template "/etc/nova/nova.conf" do
+    mode "0640"
+    source "control/nova.conf.erb"
+    variables({
+        admin_password: node[:calico][:admin_password],
+        live_migreate: node[:calico][:live_migrate]
+    })
+    owner "nova"
+    group "nova"
+    notifies :install, "package[nfs-kernel-server]", :immediately
```
Comment (alexaltair) [inline cookbooks/calico/recipes/control.rb:384]: Are these triggered by the `:live_migrate` flag? I can't infer that from the structure.

File: cookbooks/calico/recipes/general_compute.rb [hunk 4]
```diff
@@ -49,6 +51,20 @@
     sysctl net.ipv6.conf.eth0.forwarding=0
     EOH
 end
+ruby_block "persist-sysctl" do
```
Comment (alexaltair) [inline cookbooks/calico/recipes/general_compute.rb]: We don't have an "all_nodes.rb" recipe to put this in?

File: cookbooks/calico/recipes/control.rb (hunk 3 above)
Comment (robbrockbank) [inline cookbooks/calico/recipes/control.rb:384]: nope.  the live migrate stuff is part of teh nfs-kernel server

File: cookbooks/calico/recipes/general_compute.rb (hunk 4 above)
Comment (robbrockbank) [inline cookbooks/calico/recipes/general_compute.rb]: that might be what default.rb is for - but i don't want to spend time investigating just at the moment.  A lot of the compute/control nodes could be sorted out to share code

**Pull request merged**